Core/Src/shep_tasks.c

```diff
@@ -0,0 +1,223 @@
+					  .priority = osPriorityNormal };
+void vCalcNoise(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_NOISE_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		calc_noise_volt_percent(bmsdata);
+		osDelay(1000);
+	}
+}
+
+osThreadId_t calc_volt_stats_thread;
+const osThreadAttr_t calc_volt_stats_attrs = { .name = "Calc Volt Stats",
+					       .stack_size = 2048,
+					       .priority = osPriorityNormal };
+void vCalcVoltageStats(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_VOLT_STATS_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		calc_pack_voltage_stats(bmsdata);
+		osThreadFlagsSet(calc_soc_thread, CALC_SOC_FLAG);
+		osThreadFlagsSet(calc_dcl_thread, CALC_DCL_FLAG1);
+		osDelay(1000);
+	}
+}
+
+osThreadId_t calc_soc_thread;
+const osThreadAttr_t calc_soc_attrs = { .name = "Calc SoC",
+					.stack_size = 2048,
+					.priority = osPriorityNormal };
+void vCalcSoC(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_SOC_FLAG, osFlagsWaitAny, osWaitForever);
+		calc_state_of_charge(bmsdata);
+		osDelay(1000);
+	}
+}
+
+osThreadId_t calc_therms_thread;
+const osThreadAttr_t calc_therms_attrs = { .name = "Calc Therms",
+					   .stack_size = 2048,
+					   .priority = osPriorityNormal };
+void vCalcTherms(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_THERMS_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		disable_therms(bmsdata);
+		calc_cell_temps(bmsdata);
+		calc_pack_temps(bmsdata);
+		osThreadFlagsSet(calc_resistance_thread, CALC_RESIST_FLAG);
+	}
+}
+
+osThreadId_t calc_resistance_thread;
+const osThreadAttr_t calc_resistance_attrs = { .name = "Calc Resistance",
+					       .stack_size = 2048,
+					       .priority = osPriorityNormal };
+void vCalcResistance(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_RESIST_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		calc_cell_resistances(bmsdata);
+		osThreadFlagsSet(calc_dcl_thread, CALC_DCL_FLAG2);
+		osThreadFlagsSet(calc_ccl_thread, CALC_CCL_FLAG);
+	}
+}
+
+osThreadId_t calc_dcl_thread;
+const osThreadAttr_t calc_dcl_attrs = { .name = "Calc DCL",
+					.stack_size = 2048,
+					.priority = osPriorityAboveNormal };
+void vCalcDCL(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+	for (;;) {
+		osThreadFlagsWait(CALC_DCL_FLAG2 + CALC_DCL_FLAG1,
+				  osFlagsWaitAll, osWaitForever);
+		calc_dcl(bmsdata);
+		calc_cont_dcl(bmsdata);
+	}
+}
+
+osThreadId_t calc_ccl_thread;
+const osThreadAttr_t calc_ccl_attrs = { .name = "Get Segment Data",
+					.stack_size = 2048,
+					.priority = osPriorityAboveNormal };
+void vCalcCCL(void *pv_params)
+{
+	acc_data_t *bmsdata = (acc_data_t *)pv_params;
+
+	for (;;) {
+		osThreadFlagsWait(CALC_CCL_FLAG, osFlagsWaitAny, osWaitForever);
```

Comment:
have made many comments at thi spoint but i just wanna hammer it - this is so much context switching for simple calculations. i dont think it helps anything